rmgpy/solver/base.pyx

```diff
@@ -1057,11 +1060,37 @@ cdef class ReactionSystem(DASx):
                 newObjects.extend([tempNewObjects[q] for q in sortedInds])
                 newObjectInds.extend([tempNewObjectInds[q] for q in sortedInds])
                 newObjectVals.extend([tempNewObjectVals[q] for q in sortedInds])
+                newObjectType.extend([tempNewObjectType[q] for q in sortedInds])
                 
                 tempNewObjects = []
                 tempNewObjectInds = []
                 tempNewObjectVals = []
-                              
+                tempNewObjectType = []
+                
+            if branchFactor != 0.0 and not firstTime:
+                #movement of reactions to core based on branching number
+                for ind,obj in enumerate(edgeReactions):
+                    BNum = branchingNums[ind]
+                    if BNum > 1:
+                        if not(obj in newObjects or obj in invalidObjects):
```

Comment:
Honestly, in these situations I prefer minimizing the number of operators ('in', 'or', 'and', 'not') involved and mine is smaller than what you suggest.  I could make it smaller by doing:
`obj not in newObjects+invalidObjects`
or 
`objs = newObjects+invalidObjects`
`obj not in objs`
but I feel that the first is undesirable for speed reasons and the second is undesirable because it introduces an new variable in simulate exclusively for the if statement.